helpers/plugins.js

```diff
@@ -0,0 +1,26 @@
+const pluginSets = new Map();
+
+export function getPlugin(setKey, pluginKey) {
+	const pluginSet = pluginSets.get(setKey);
+	return pluginSet?.plugins.find(plugin => plugin.options.key === pluginKey)?.plugin;
+}
+
+export function getPlugins(setKey) {
+	const pluginSet = pluginSets.get(setKey);
+	if (!pluginSet) return [];
+	if (pluginSet.requiresSorting) {
+		pluginSet.plugins.sort((item1, item2) => item1.options.sort - item2.options.sort);
+		pluginSet.requiresSorting = false;
+	}
+	return pluginSet.plugins.map(item => item.plugin);
+}
+
+export function registerPlugin(setKey, plugin, options) {
+	let pluginSet = pluginSets.get(setKey);
+	if (!pluginSet) {
+		pluginSet = { plugins: [], requiresSorting: false };
```

Comment:
When I originally wrote this, this wrapper object around the `plugins` array and `requiresSorting` didn't exist, and I was blindly slapping `requiresSorting` onto the native array, which felt a little wrong even though it wasn't causing any issues. 😬 An interesting behaviour that that resulted in was - if another plugin was registered after the consumer called `getPlugins`, it would magically show up in their array because they are referencing the same array.

So a consequence of this wrapper (and the `map(...)` above) is that consumers get their own array, and if plugins are registered after they called `getPlugins`, they won't see it unless they call `getPlugins` again. Feature or bug? Consumers getting their own array seems safer, and all registrations _should_ technically be done by the time anyone requests plugins. (is this assumption true?)

Reply:
I think consumers having their own copy is good, since theoretically they could sort it themselves in a different way (e.g. alphabetically) by something only they understand and we wouldn't want that impacting others.

Reply:
That's a good point.